ui/app/store/actions.js

```diff
@@ -2210,9 +2211,19 @@ export function setPendingTokens(pendingTokens) {
   const { address, symbol, decimals } = customToken
   const tokens =
     address && symbol && decimals
-      ? { ...selectedTokens, [address]: { ...customToken, isCustom: true } }
+      ? {
+          ...selectedTokens,
+          [address]: {
+            ...customToken,
+            isCustom: true,
+          },
+        }
       : selectedTokens
 
+  Object.keys(tokens).forEach((tokenAddress) => {
+    tokens[tokenAddress].unlisted = contractMap[tokenAddress] === undefined
```

Comment:
We should normalize this token address - to lowercase perhaps? I believe we accept any case as input, but the `eth-contract-metadata` entries are mostly mixed-case (check-summed, as specified in [EIP-55](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md)).

The same goes for the suggested tokens from dapps - I don't know that we enforce case there either.

Reply:
so is your suggestion:
```js
  Object.keys(tokens).forEach((tokenAddress) => {
    tokens[tokenAddress.toLowerCase()].unlisted = contractMap[tokenAddress] === undefined
```
If so, should we also do:
```js
const contractMapLowerCased = contractMap.reduce((acc, { address, token }) => {
  acc[address.toLowerCase()] = token
  return acc
}, {})

// ... later ...

  Object.keys(tokens).forEach((tokenAddress) => {
    tokens[tokenAddress.toLowerCase()].unlisted = contractMapLowerCased[tokenAddress.toLowerCase()] === undefined
```

so that we are not marking things unlisted that are just mismatch case?

Reply:
Yep, something like that would certainly work!

Possible simpler alternative might be

```
  const listedContractAddresses = contractMap.map(address => address.toLowerCase())

  // ... later ...

  Object.keys(tokens).forEach((tokenAddress) => {
    tokens[tokenAddress].unlisted = !contractMapLowerCased.includes(tokenAddress.toLowerCase())
```

Reply:
Check out https://github.com/MetaMask/metamask-extension/pull/9810/commits/3f7da94b790796e20ef2ed97e5950be8bf2e0d85 and lemme know what you think.

Reply:
Looks good!